modules/minimint-wallet/src/lib.rs

```diff
@@ -1095,33 +1096,32 @@ impl<'a> StatelessWallet<'a> {
                 secret_key
             };
 
-            let tx_hash = tx_hasher.signature_hash(
-                idx,
-                psbt_input
-                    .witness_script
-                    .as_ref()
-                    .expect("Missing witness script"),
-                psbt_input
-                    .witness_utxo
-                    .as_ref()
-                    .expect("Missing UTXO")
-                    .value,
-                SigHashType::All,
-            );
+            let tx_hash = tx_hasher
+                .segwit_signature_hash(
+                    idx,
+                    psbt_input
+                        .witness_script
+                        .as_ref()
+                        .expect("Missing witness script"),
+                    psbt_input
+                        .witness_utxo
+                        .as_ref()
+                        .expect("Missing UTXO")
+                        .value,
+                    EcdsaSighashType::All,
+                )
+                .expect("Failed to create segwit sighash");
```

Comment:
This `expect` is a little strange but we're doing it a bunch of times in this method. I guess we're just assuming that something is horribly wrong if we can't produce a sig hash?

Reply:
I only see two potential errors:
* IO/write errors in `segwit_encode_signing_data_to`, but since `segwit_signature_hash` writes to a infallible hasher that's not a concern
* [`IndexOutOfInputsBounds` in `segwit_encode_signing_data_to`](https://docs.rs/bitcoin/0.28.1/src/bitcoin/util/sighash.rs.html#582) if the input index is wrong